src/Agent.Listener/Configuration/InteractiveSessionConfigurationManager.cs

```diff
@@ -0,0 +1,243 @@
+#if OS_WINDOWS
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(InteractiveSessionConfigurationManager))]
+    public interface IInteractiveSessionConfigurationManager : IAgentService
+    {
+        void Configure(CommandSettings command);
+        void UnConfigure(int listenerProcessId);
+        bool RestartNeeded();
+        bool IsInteractiveSessionConfigured();
+    }
+
+    public class InteractiveSessionConfigurationManager : AgentService, IInteractiveSessionConfigurationManager
+    {
+        private ITerminal _terminal;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+            _terminal = hostContext.GetService<ITerminal>();
+        }
+
+        public void Configure(CommandSettings command)
+        {
+            var logonAccount = command.GetAutoLogonUserName();
+
+            string domainName;
+            string userName;
+
+            GetAccountSegments(logonAccount, out domainName, out userName);
+
+            if ((string.IsNullOrEmpty(domainName) || domainName.Equals(".", StringComparison.CurrentCultureIgnoreCase)) && !logonAccount.Contains("@"))
+            {
+                logonAccount = String.Format("{0}\\{1}", Environment.MachineName, userName);
+            }
+
+            Trace.Info("LogonAccount after transforming: {0}, user: {1}, domain: {2}", logonAccount, userName, domainName);
+
+            string logonPassword = string.Empty;
+            var windowsServiceHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            while (true)
+            {
+                logonPassword = command.GetWindowsLogonPassword(logonAccount);
+                if (windowsServiceHelper.IsValidCredential(domainName, userName, logonPassword))
+                {
+                    Trace.Info("Credential validation succeeded");
+                    break;
+                }
+                
+                if (command.Unattended)
+                {
+                    throw new SecurityException(StringUtil.Loc("InvalidWindowsCredential"));
+                }
+                    
+                Trace.Info("Invalid credential entered");
+                _terminal.WriteLine(StringUtil.Loc("InvalidWindowsCredential"));
+            }
+
+            bool isCurrentUserSameAsAutoLogonUser = windowsServiceHelper.IsTheSameUserLoggedIn(domainName, userName);                
+            var securityIdForTheUser = windowsServiceHelper.GetSecurityIdForTheUser(userName);
```

Comment:
thanks. Didnt realize about the domain name. Done nw.

---
In reply to: [117657318](https://github.com/Microsoft/vsts-agent/pull/978#discussion_r117657318) [](ancestors = 117657318)